File: Answers/07_while/e12_primeNumbers.cs
using System;

class Program
{
    static void Main()
    {
        Console.WriteLine(@"This program will find the prime numbers for you. You can choose:
1- just the prime numbers
2- numbers that are not prime numbers
3- both
q- quit");

        while (true)
        {
            string choice = Console.ReadLine();
            if (choice == "q")
                break;

            if (choice == "1" || choice == "2" || choice == "3")
            {
                int upper = int.Parse(Console.ReadLine());

                for (int num = 2; num <= upper; num++)
                {
                    bool isPrime = true;
                    for (int i = 2; i <= Math.Sqrt(num); i++)
                    {
                        if (num % i == 0)
                        {
                            isPrime = false;
                            break;
                        }
                    }

                    if (choice == "1" && isPrime)
                    {
                        Console.WriteLine($"{num}, is a prime number");
                    }
                    else if (choice == "2" && !isPrime)
                    {
                        for (int i = 2; i < num; i++)
                        {
                            if (num % i == 0)
                            {
                                Console.WriteLine($"{num} equals {i}*{num / i}");
                                break;
                            }
                        }
                    }
                    else if (choice == "3")
                    {
                        if (isPrime)
                        {
                            Console.WriteLine($"{num}, is a prime number");

File: Exercies/07_while/e12_primeNumbers.py
from sympy import isprime

print("""This program will find the prime numbers for you. You can choose:
1- just the prime numbers
2- numbers that are not prime numbers
3- both
q- quit""")

while True:
    choice = input("- ")
    if choice == 'q':
        break

    if choice in ['1', '2', '3']:
        upper = int(input("Give the upper border: "))

        if choice == '1':
            for num in range(2, upper + 1):
                if isprime(num):
                    print(f"{num}, is a prime number")
        elif choice == '2':
            for num in range(2, upper + 1):
                if not isprime(num):
                    for i in range(2, num):
                        if num % i == 0:
                            print(f"{num} equals {i}*{num // i}")
                            break
        elif choice == '3':
            for num in range(2, upper + 1):
                if isprime(num):
                    print(f"{num}, is a prime number")
                else:
                    for i in range(2, num):
                        if num % i == 0:
                            print(f"{num} equals {i}*{num // i}")
                            break
    else:
        print("Invalid choice. Please enter 1, 2, 3, or q.")
